flox/data/core.py

```diff
@@ -1,12 +1,19 @@
+from __future__ import annotations
+
 import abc
-from typing import TypeVar, Iterable
+import typing
 
 from torch.utils.data import Dataset, Subset
 
-from flox.flock import FlockNodeID, FlockNode
-from flox.flock.states import NodeState
+from flox.flock import FlockNode
+
+if typing.TYPE_CHECKING:
+    from typing import TypeVar, Iterator
+
+    from flox.flock.states import NodeState
+    from flox.flock import FlockNodeID
```

Comment:
any reason these can't be imported at runtime too? (and same for other places we do `if typing.TYPE_CHECKING`)

Reply:
Personally, I think there's a bit of added clarity for which parts are actual runtime dependencies as opposed to just being included for type hinting. There's also just the issue of not wanting to unintentionally and unnecessarily importing `torch` because you are importing a `flox` module that uses `FloxModule` as a type-hint without instantiating one.

Reply:
makes sense, especially re: torch 👍